modules/FindBasisUniversal.cmake

```diff
@@ -161,30 +163,71 @@ foreach(_component ${BasisUniversal_FIND_COMPONENTS})
                     ${BasisUniversalEncoder_DIR}/basisu_kernels_sse.cpp
                     ${BasisUniversalEncoder_DIR}/basisu_pvrtc1_4.cpp
                     ${BasisUniversalEncoder_DIR}/basisu_resampler.cpp
                     ${BasisUniversalEncoder_DIR}/basisu_resample_filters.cpp
                     ${BasisUniversalEncoder_DIR}/basisu_ssim.cpp
-                    ${BasisUniversalEncoder_DIR}/basisu_uastc_enc.cpp
-                    ${BasisUniversalEncoder_DIR}/jpgd.cpp
-                    ${BasisUniversalEncoder_DIR}/lodepng.cpp)
+                    ${BasisUniversalEncoder_DIR}/basisu_uastc_enc.cpp)
+
+                # Files not present in all supported basis versions, treat them
+                # as optional and do nothing if not found.
+                foreach(_file
+                    # Removed in 1.16
+                    basisu_astc_decomp.cpp
+                    basisu_global_selector_palette_helpers.cpp
+                    # Added in 1.16
+                    basisu_opencl.cpp
+                    # Added in 1.50
+                    basisu_astc_hdr_enc.cpp
+                    3rdparty/android_astc_decomp.cpp)
+                    # Disable the find root path here, it overrides the
+                    # CMAKE_FIND_ROOT_PATH_MODE_INCLUDE setting potentially set in
+                    # toolchains.
+                    find_file(BasisUniversalEncoder_${_file}_SOURCE NAMES ${_file}
+                        HINTS ${BasisUniversalEncoder_DIR} NO_CMAKE_FIND_ROOT_PATH)
+
+                    if(BasisUniversalEncoder_${_file}_SOURCE)
+                        list(APPEND BasisUniversalEncoder_SOURCES
+                            ${BasisUniversalEncoder_${_file}_SOURCE})
+                    endif()
+                endforeach()
+
+                # Source files for image file loading are not added here.
+                # BasisImageConverter provides stubs for all used functions.
+                # See src/external/basis-uncrapifier for more info.
 
                 foreach(_file ${BasisUniversalEncoder_SOURCES})
                     _basis_setup_source_file(${_file})
                 endforeach()
 
+                set(BasisUniversalEncoder_DEFINITIONS "BASISU_NO_ITERATOR_DEBUG_LEVEL")
+
+                # Basis already checks for GCC < 5 to work around missing
+                # std::is_trivially_copyable but then uses the wrong built-in.
+                # So close, yet so far.
+                # Needs to be transitive because it's used in a header file.
+                if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5.0")
+                    list(APPEND BasisUniversalEncoder_DEFINITIONS
+                        "__is_trivially_copyable=__has_trivial_copy")
+                endif()
+
+                # Try to find an external OpenCL library and enable support for
+                # it in basis if found.
+                find_package(OpenCL QUIET)
```

Comment:
I'm thinking this should probably not be `QUIET`, same as Zstd is not `QUIET` either.